deob-bytecode/src/main/kotlin/org/openrs2/deob/bytecode/transform/StringDecryptionTransformer.kt

```diff
@@ -0,0 +1,183 @@
+
+                val xorLookup = ArrayList<Number>()
+                for (label in switch.labels) {
+                    xorLookup.add(label.next.intConstant ?: continue)
+                }
+                xorLookup.add(switch.dflt.next.intConstant ?: continue)
+
+                // step 3: decrypt all strings in the initializer
+                val strings = ArrayList<String>()
+                var field: FieldNode? = null
+                var fieldInit: AbstractInsnNode? = null
+                for (insn in clinit.instructions) {
+                    if (insn.opcode != Opcodes.INVOKESTATIC) continue
+                    val invoke = insn as MethodInsnNode
+                    if (invoke.owner != clazz.name || invoke.name != outer.name || invoke.desc != outer.desc) continue
+
+                    val ldc = invoke.previous.previous as LdcInsnNode
+                    val str = ldc.cst as String
+                    val decrypted = decryptString(str, xorLookup)
+
+                    ldc.cst = decrypted
+                    strings.add(decrypted)
+                    stringsDecrypted++
+
+                    if (field == null) {
+                        if (invoke.next?.opcode == Opcodes.AASTORE && invoke.next?.next?.opcode == Opcodes.PUTSTATIC) {
+                            // this one is the static String[] field
+                            val put = invoke.next.next as FieldInsnNode
+                            if (put.owner == clazz.name && put.desc == "[Ljava/lang/String;") {
+                                field = clazz.fields.find { it.name == put.name }
+                                fieldInit = put
+                            }
+                        } else if (invoke.next?.opcode == Opcodes.PUTSTATIC) {
+                            // this one is the static String field
+                            val put = invoke.next as FieldInsnNode
+                            if (put.owner == clazz.name && put.desc == "Ljava/lang/String;") {
+                                field = clazz.fields.find { it.name == put.name }
+                                fieldInit = put
+                            }
+                        }
+                    }
+
+                    clinit.instructions.remove(invoke.previous)
+                    clinit.instructions.remove(invoke)
+                }
+
+                // step 4: inline all string references to the static String[] field (specifically static!)
+                if (field != null) {
+                    for (method in clazz.methods) {
+                        if (field.desc == "[Ljava/lang/String;") {
+                            // multiple encrypted strings in the class so an array is produced
+                            for (insn in method.instructions) {
+                                // we're looking for getstatic -> constant -> aaload
+                                // we start backwards (aaload) and check the previous 2 instructions
```

Comment:
Again - I think this indicates we should use InsnMatcher here - something like `GETSTATIC (ICONST | BIPUSH | SIPUSH | LDC) AALOAD` for the regex?